agent_plugin_builder/agent_plugin_builder.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from argparse import ArgumentParser
+from pathlib import Path
+from tempfile import mkdtemp
+
+from .build_plugin import BUILD, DIST, build_agent_plugin
+from .setup_build_plugin_logging import (
+    AGENT_PLUGIN_BUILDER_LOG_FILENAME,
+    reset_logger,
+    setup_logging,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def main():
+    parser = ArgumentParser(description="Build plugin")
+    parser.add_argument("plugin_path", metavar="PLUGIN_PATH", type=Path, help="Path to the plugin)")
+    parser.add_argument(
+        "-b",
+        "--build_dir_path",
+        metavar="BUILD_DIR_PATH",
+        type=Path,
+        default=(Path.cwd() / BUILD),
+        help="Optional path to the build directory. Default: <current_working_directory>/build.",
+    )
+    parser.add_argument(
+        "-d",
+        "--dist_dir_path",
+        metavar="DIST_DIR_PATH",
+        type=Path,
+        default=(Path.cwd() / DIST),
+        help="Optional path to the dist directory. Default: <current_working_directory>/dist.",
+    )
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        dest="verbosity",
+        action="count",
+        default=-1,
+        help="Verbosity (between 1-5 occurrences with more leading to more "
+        "verbose logging). CRITICAL=1, ERROR=2, WARN=3, INFO=4, "
+        "DEBUG=5. Default(0 or more than 5 occurrences)  level: INFO.",
+    )
+    args = parser.parse_args()
+    _temp_log_dir = Path(mkdtemp())
+    _setup_logging(_temp_log_dir, args.verbosity)
+    _log_arguments(args)
+    try:
+        build_agent_plugin(args.plugin_path, args.build_dir_path, args.dist_dir_path)
+    except Exception as e:
+        logger.error(f"Error building plugin: {e}", exc_info=True)
+
+    logger.info(f"Copying log file to {args.build_dir_path}")
+    import shutil
+
+    shutil.copy(_temp_log_dir / AGENT_PLUGIN_BUILDER_LOG_FILENAME, args.build_dir_path)
```

Comment:
I'd recommend we not write anything to /tmp. Anything we need to output before the log file is set up can be output to STDERR. Anything else can be output after the log file is created.